pkg/controller/deployment/recreate.go

```diff
@@ -104,8 +106,19 @@ func oldPodsRunning(newRS *extensions.ReplicaSet, oldRSs []*extensions.ReplicaSe
 		if newRS != nil && newRS.UID == rsUID {
 			continue
 		}
-		if len(podList.Items) > 0 {
-			return true
+		for _, pod := range podList.Items {
+			switch pod.Status.Phase {
+			case v1.PodFailed, v1.PodSucceeded:
```

Comment:
It should be impossible for RS to have succeeded pods, but there's no harm to add this check.

Reply:
It should be impossible, but a prefer to handle even such case explicitly.